### PR TITLE
Implement app service V2 API test cases

### PR DESCRIPTION
**Pull request by jinlinGuan**
Fix https://github.com/edgexfoundry/edgex-taf/issues/155

Signed-off-by: Ginny Guan <ginny@iotechsys.com>

### CODE REVIEW

File: TAF/testData/app-service/trigger_data.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      {
+        "id": "5bd794de0e36080001f53eac",
+        "origin": 1540855006469,
+        "device": "Random-Float-Device",
+        "name": "RandomValue_Float64",
+        "value": "invalid value"
+      }
+    ]
+  },
+  "Unprocessable Entity": {
```
Comment (lenny-intel) [inline TAF/testData/app-service/trigger_data.json:38]: Why is this `Unprocessable`? That added property should just be ignored.

File: TAF/testScenarios/functionalTest/V2-API/app-service/info/GET.robot [hunk 2]
```diff
@@ -1,28 +1,37 @@
...
+...                            AND  Run Keyword if  $SECURITY_SERVICE_NEEDED == 'true'  Remove Token
 
 *** Variables ***
 ${SUITE}          App-Service GET Testcases
 ${LOG_FILE_PATH}  ${WORK_DIR}/TAF/testArtifacts/logs/app-service-get.log
 ${edgex_profile}  blackbox-tests
+${AppServiceUrl_blackbox}  http://${BASE_URL}:48095
+${api_version}  v2
 
 *** Test Cases ***
```
Comment (lenny-intel) [inline TAF/testScenarios/functionalTest/V2-API/app-service/info/GET.robot:17]: All response V2 DTOs now have apiVersion. Should be verifying that it is present?

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/functionalTest/V2-API/app-service/info/GET.robot (hunk 2 above)
Comment (jinlinGuan) [reply TAF/testScenarios/functionalTest/V2-API/app-service/info/GET.robot:15]: Added apiVersion check.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by jinlinGuan** [comment]
The changes: 

1. update the sed command to correctly insert environment variable
 ```
  if [ "$USE_SECURITY" = '-security-' ]; then
    # this sed command will fail on Mac
    sed -i '/hostname: edgex-vault-worker/i \
      ADD_SECRETSTORE_TOKENS: appservice-http-export-secrets ' docker-compose.yaml
  fi
```
2. replace `Default Tags daily-run` with `Default Tags v2-api`
3. add `Check app-service is available` keyword to `Setup Suite for App Service`

File: TAF/testCaseModules/keywords/app-service/AppServiceAPI.robot [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Library   OperatingSystem
+Library   TAF/testCaseModules/keywords/setup/edgex.py
+Library   TAF/testCaseModules/keywords/setup/startup_checker.py
+Library   TAF/testCaseModules/keywords/consul/consul.py
+Resource  TAF/testCaseModules/keywords/common/commonKeywords.robot
+
+*** Keywords ***
+Setup Suite for App Service
+    [Arguments]  ${appServiceUrl}
+    Run Keywords  Setup Suite  AND  Set Suite Variable  ${url}  ${appServiceUrl}
```
[the remark this answers is not in the document]
Comment (jinlinGuan) [reply TAF/testCaseModules/keywords/app-service/AppServiceAPI.robot]: Changes made.

**Comment by jinlinGuan** [comment]
add keyword `Should Return Status Code "500"` to secrets/POST.robt SecretsPOST001 and SecretsPOST002 testcases for non-security test, and remove `Run Keyword if  $SECURITY_SERVICE_NEEDED == 'true'` in missing key and missing value testcases(ErrSecretsPOST001 and ErrSecretsPOST002), because that check has priority over the security secret store check.

**Review by cherrycl — APPROVED** [review]
LGTM

**Pull request merged**